### PR TITLE
Added "<param>*2/2" check in Test SQL Injection

### PR DESCRIPTION
**Pull request by rnehra01**
Added *<param>*2/2* check to the Logic based ones (for integer parameter 
 * values).. if the result is the same, it might be a SQL Injection 
It was mentioned as *TODO*. It's done.

### CODE REVIEW

File: src/org/zaproxy/zap/extension/ascanrules/TestSQLInjection.java [hunk 1]
```diff
@@ -719,7 +719,7 @@ public void scan(HttpMessage msg, String param, String origParamValue) {
 					String expressionBodyOutput[] = {modifiedExpressionOutputUnstripped, modifiedExpressionOutputStripped};
 					boolean strippedOutput[] = {false, true};
 
-					 for (int booleanStrippedUnstrippedIndex = 0; booleanStrippedUnstrippedIndex < 2 && !sqlInjectionFoundForUrl; booleanStrippedUnstrippedIndex++) {
+					for (int booleanStrippedUnstrippedIndex = 0; booleanStrippedUnstrippedIndex < 2 && !sqlInjectionFoundForUrl; booleanStrippedUnstrippedIndex++) {
```
Comment (kingthorin) [inline src/org/zaproxy/zap/extension/ascanrules/TestSQLInjection.java]: Formatting change should be submitted as a separate PR: https://github.com/zaproxy/zaproxy/wiki/DevGuidelines#formatting

File: src/org/zaproxy/zap/extension/ascanrules/TestSQLInjection.java [hunk 2]
```diff
@@ -779,6 +779,98 @@ public void scan(HttpMessage msg, String param, String origParamValue) {
 							return;
 						}
 					}
+					// Check 4a
+					// this check is to see whwter the "1" will give the same result as "1*2/2"
```
Comment (kingthorin) [inline src/org/zaproxy/zap/extension/ascanrules/TestSQLInjection.java]: `whether`

File: src/org/zaproxy/zap/extension/ascanrules/TestSQLInjection.java [hunk 3]
```diff
@@ -779,6 +779,98 @@ public void scan(HttpMessage msg, String param, String origParamValue) {
...
+
+					if (!sqlInjectionFoundForUrl && countExpressionBasedRequests < doExpressionMaxRequests){
+						//get a value 2 times bigger
+						int paramtwice = paramAsInt * 2;
```
Comment (kingthorin) [inline src/org/zaproxy/zap/extension/ascanrules/TestSQLInjection.java]: Might need to ensure this doesn't exceed _maxint_

File: src/org/zaproxy/zap/extension/ascanrules/TestSQLInjection.java [hunk 4]
```diff
@@ -779,6 +779,98 @@ public void scan(HttpMessage msg, String param, String origParamValue) {
...
+						String modifiedParamValue4a = String.valueOf(paramtwice) + "/2";
+
+						//and prepare a request to set the parameter value to a string value like "2/2", if the original parameter value was "1"
+						//those of you still paying attention will note that if handled as expressions (such as by a database), these represent the same value.
```
Comment (kingthorin) [inline src/org/zaproxy/zap/extension/ascanrules/TestSQLInjection.java]: This logic won't always work will it? If paramAsInt is say 3, now you have 6/2 which is not 1...

**Comment by kingthorin** [comment]
Also, please do an interactive rebase and fix-up or squash into a single "atomic" commit: https://github.com/zaproxy/zaproxy/blob/develop/CONTRIBUTING.md#what-we-zap-team-expect-from-you .

- https://www.kernel.org/pub/software/scm/git/docs/git-rebase.html#_interactive_mode
- https://wiki.eclipse.org/EGit/User_Guide#Interactive_Rebase

File: src/org/zaproxy/zap/extension/ascanrules/TestSQLInjection.java (hunk 4 above)
Comment (rnehra01) [inline src/org/zaproxy/zap/extension/ascanrules/TestSQLInjection.java]: The idea is to keep the parameter value **same**, not to make it 1.

Comment (kingthorin) [inline src/org/zaproxy/zap/extension/ascanrules/TestSQLInjection.java]: Stick with the example given. If the value is 1 you want it to become 2/2 which evaluates to 1. However if you double the original and then always assume /2 you won't always get 1.

The line should be?
`String modifiedParamValue4a = String.valueOf(paramtwice) + "/" + String.valueOf(paramtwice);`

Comment (kingthorin) [inline src/org/zaproxy/zap/extension/ascanrules/TestSQLInjection.java]: I believe the original expression test is flawed.... (The +2 variant)

Comment (kingthorin) [inline src/org/zaproxy/zap/extension/ascanrules/TestSQLInjection.java]: OMG ignore me....at 01:35 it seems I'm having a brain fart or something.....sigh

**Comment by rnehra01** [comment]
I don't understand the **Strength guidelines** thing mentioned above. 
I could only tell, if **attack strength** is **MEDIUM** or above, only then this check (4a) will be performed.

**Comment by kingthorin** [comment]
I just wondered if adding the extra 2 requests for test 4a moved the expression tests beyond any of the strength boundaries.

**Comment by kingthorin** [comment]
Hopefully another PR is coming with these changes all fixed-up/squashed?

Let us know if you need help with the interactive rebase.

**Comment by thc202** [comment]
Just for the record, the follow up pull request is #706.
